packages/react-devtools-core/src/cachedSettings.js

```diff
@@ -0,0 +1,96 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import {
+  type ConsolePatchSettings,
+  writeConsolePatchSettingsToWindow,
+  patchConsoleUsingWindowValues,
+} from 'react-devtools-shared/src/backend/console';
+import {
+  LOCAL_STORAGE_SHOULD_APPEND_COMPONENT_STACK_KEY,
+  LOCAL_STORAGE_SHOULD_BREAK_ON_CONSOLE_ERRORS,
+  LOCAL_STORAGE_SHOW_INLINE_WARNINGS_AND_ERRORS_KEY,
+  LOCAL_STORAGE_HIDE_CONSOLE_LOGS_IN_STRICT_MODE,
+  LOCAL_STORAGE_BROWSER_THEME,
+} from 'react-devtools-shared/src/constants';
+import type {BrowserTheme} from 'react-devtools-shared/src/devtools/views/DevTools';
+
+export type CachedSettingsStore = {
+  setValue: (value: string) => void,
+  getValue: () => ?string,
+};
+
+export type CachedSettings = {
+  consolePatchSettings: ?ConsolePatchSettings,
+};
+
+export function initializeUsingCachedSettings(
+  cachedSettingsStore: CachedSettingsStore,
+) {
+  const cachedSettingsString = cachedSettingsStore.getValue();
+  if (cachedSettingsString == null) {
+    return;
+  }
+  const cachedSettings = parseCachedSettings(cachedSettingsString);
+  if (cachedSettings != null && cachedSettings.consolePatchSettings != null) {
+    writeConsolePatchSettingsToWindow(cachedSettings.consolePatchSettings);
+  }
+}
+
+function parseCachedSettings(cachedSettingsString: string): ?CachedSettings {
+  const parsedValue = JSON.parse(cachedSettingsString);
```

Comment:
This is all wrapped in a try/catch in src/backend. I wrapped it once (there) instead of in here because there are actually two things that can throw:

* `const cachedSettingsString = cachedSettingsStore.getValue();` (because `cachedSettingsStore` is untrusted)
* `JSON.parse`

But anyway, happy to wrap it individually if that seems more elegant.

(Also, `cachedSettingsStore.setValue` can throw, but that happens to not cause any negative outcomes)

One can also cause this to throw by causing `get cachedSettingsStore` to throw or by lying with flowtypes... I'm not counting those.